File: dotnet/system/protocol/allors.protocol.json/api/pull/PullResponse.cs
// <copyright file="PullResponse.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Json.Api.Pull
{
    using System.Collections.Generic;
    using System.Text.Json.Serialization;

    public class PullResponse : Response
    {
        [JsonPropertyName("c")]
        public IDictionary<string, long[]> Collections { get; set; }

        [JsonPropertyName("o")]
        public IDictionary<string, long> Objects { get; set; }

        [JsonPropertyName("v")]
        public IDictionary<string, object> Values { get; set; }

        [JsonPropertyName("p")]
        public PullResponseObject[] Pool { get; set; }

        [JsonPropertyName("a")]
        public long[][] AccessControls { get; set; }
    }
}

File: typescript/modules/libs/system/common/protocol-json/src/lib/api/pull/pull-response.ts
import { PullResponseObject } from './pull-response-object';
import { Response } from '../response';

export interface PullResponse extends Response {
  /** Collections */
  c: { [name: string]: number[] };

  /** Objects */
  o: { [name: string]: number };

  /** Values */
  v: { [name: string]: string };

  /** Pool */
  p: PullResponseObject[];

  /** Grants [id, version] */
  g: number[][];

  /** Revocations [id, version] */
  r: number[][];
}
